imblearn/over_sampling/_random_over_sampler.py

```diff
@@ -148,34 +147,41 @@ def _check_X_y(self, X, y):
     def _fit_resample(self, X, y):
         random_state = check_random_state(self.random_state)
 
-        if self.smoothed_bootstrap:
-            if isinstance(self.shrinkage, Real):
-                self.shrinkage_ = {
-                    klass: self.shrinkage for klass in self.sampling_strategy_
-                }
-            else:
-                missing_shrinkage_keys = (
-                    self.sampling_strategy_.keys() - self.shrinkage.keys()
+        if isinstance(self.shrinkage, Real):
+            self.shrinkage_ = {
+                klass: self.shrinkage for klass in self.sampling_strategy_
+            }
+        else:
```

Comment:
probably `elif` here. Otherwise we should raise `ValueError`?

Reply:
I don't like the `elif` here because I would need to check for `None` and `dict` but a we would be fine accepting something that behave like a `dict`. I prefer to be permissive :)

Reply:
Sorry. I mean if you pass a list you will get an unhandled exception like `AttributeError: 'list' object has no attribute 'keys'`. You need to ensure that is dict like or `Number`

Reply:
OK, so let's ducktype then :)